hessian/src/main/java/com/linecorp/armeria/server/hessian/HessianHttpServiceBuilder.java

```diff
@@ -0,0 +1,172 @@
+            defaultExceptionHandler = (
+            ctx, cause) -> RpcResponse.ofFailure(cause);
+
+    private final ImmutableList.Builder<ServiceEntry> hessianService = ImmutableList.builder();
+
+    @Nullable
+    private Function<? super RpcService, ? extends RpcService> decoratorFunction;
+
+    private BiFunction<? super ServiceRequestContext, ? super Throwable, ? extends RpcResponse>
+            exceptionHandler = defaultExceptionHandler;
+
+    HessianHttpServiceBuilder() {
+    }
+
+    /**
+     * Add a hessian service.
+     * @param path service path.
+     * @param apiClass hessian api.
+     * @param implementation hessian implementation.
+     * @param blocking blocking or not.
+     */
+    public HessianHttpServiceBuilder addService(String path, Class<?> apiClass, Object implementation,
+                                                boolean blocking) {
+        requireNonNull(path, "path");
+        requireNonNull(apiClass, "apiClass");
+        requireNonNull(implementation, "implementation");
+        hessianService.add(new ServiceEntry(path, apiClass, implementation, blocking));
+        return this;
+    }
+
+    /**
+     * Add a hessian service.
+     * @param path service path.
+     * @param apiClass hessian api.
+     * @param implementation hessian implementation.
+     */
+    public HessianHttpServiceBuilder addService(String path, Class<?> apiClass, Object implementation) {
+        return addService(path, apiClass, implementation, true);
+    }
+
+    /**
+     * Sets the {@link BiFunction} that returns an {@link RpcResponse} using the given
+     * {@link Throwable} and {@link ServiceRequestContext}.
+     */
+    public HessianHttpServiceBuilder exceptionHandler(
+            BiFunction<? super ServiceRequestContext, ? super Throwable, ? extends RpcResponse>
+                    exceptionHandler) {
+        this.exceptionHandler = requireNonNull(exceptionHandler, "exceptionHandler");
+        return this;
+    }
+
+    /**
+     * A {@code Function<? super RpcService, ? extends RpcService>} to decorate the
+     * {@link RpcService}.
+     */
+    public HessianHttpServiceBuilder decorate(
+            Function<? super RpcService, ? extends RpcService> decoratorFunction) {
+        requireNonNull(decoratorFunction, "decoratorFunction");
+        if (this.decoratorFunction == null) {
+            this.decoratorFunction = decoratorFunction;
+        } else {
+            this.decoratorFunction = this.decoratorFunction.andThen(decoratorFunction);
+        }
+        return this;
+    }
+
+    private RpcService decorate(RpcService service) {
+        if (decoratorFunction != null) {
+            return service.decorate(decoratorFunction);
+        }
+        return service;
+    }
+
+    /**
+     * Builds a new instance of {@link HessianHttpServiceImpl}.
+     */
+    public HessianHttpServiceImpl build() {
```

Comment:
We can't expose `HessianHttpServiceImpl` in this public API because `HessianHttpServiceImpl` is an internal class. How about returning `HessianHttpService` instead?